src/test/scala/edu/ie3/mobsim/model/ElectricVehicleSpec.scala

```diff
@@ -272,10 +265,92 @@ class ElectricVehicleSpec extends UnitSpec with ElectricVehicleTestData {
         }
       }
     }
+
+    "An electricVehicle" should {
+      "check if home charging is possible" in {
+        evWithHomeCharging.chargingAtHomePossible shouldBe true
+        evWithoutHomeCharging.chargingAtHomePossible shouldBe false
+      }
+
+      "copy object with new stored energy" in {
+        val evFull: ElectricVehicle =
+          evWithHomeCharging.copyWith(evWithHomeCharging.getStoredEnergy)
+        evFull.getStoredEnergy shouldBe givenModel.capacity
+
+        val zero: ComparableQuantity[Energy] =
+          Quantities.getQuantity(0, PowerSystemUnits.KILOWATTHOUR)
+        val evEmpty: ElectricVehicle = evWithHomeCharging.copyWith(zero)
+        evEmpty.getStoredEnergy shouldBe zero
+      }
+
+      "copy object with updated charging at simona information" in {
+        val evChargingAtSimona: ElectricVehicle =
+          evWithHomeCharging.setChargingAtSimona()
+        evChargingAtSimona.chargingAtSimona shouldBe true
+
+        val evNotChargingAtSimona: ElectricVehicle =
+          evWithHomeCharging.removeChargingAtSimona()
+        evNotChargingAtSimona.chargingAtSimona shouldBe false
+      }
+
+      "copy object with chosen charging station" in {
+        val evSetChargingStation: ElectricVehicle =
+          evWithHomeCharging.setChosenChargingStation(Some(cs6.uuid))
+        evSetChargingStation.chosenChargingStation shouldBe Some(cs6.uuid)
+
+        val evNoChargingStation: ElectricVehicle =
+          evWithHomeCharging.setChosenChargingStation(None)
+        evNoChargingStation.chosenChargingStation shouldBe None
+      }
+
+      "return the correct poiType for the destinationPoi" in {
+        val ev: ElectricVehicle = ev1.copyWith(
+          storedEnergy = ev1.storedEnergy,
+          bbpgPoi,
+          parkingTimeStart = ev1.parkingTimeStart,
+          departureTime = ev1.departureTime
+        )
+
+        ev.getDestinationPoiType shouldBe bbpgPoi.getPoiType
+      }
+
+      "return the correct departure tick" in {
+        val time: ZonedDateTime = ZonedDateTime.now().plusHours(1)
+
+        val ev: ElectricVehicle = ev1.copyWith(
+          storedEnergy = ev1.storedEnergy,
+          destinationPoi = ev1.destinationPoi,
+          parkingTimeStart = ev1.parkingTimeStart,
+          time
+        )
+
+        ev.departureTime shouldBe time
+      }
+
+      "update charging price memory correctly" in {
+        val evNoQueue: ElectricVehicle = ev1
+        val queue: immutable.Queue[Double] = immutable.Queue.empty
+
+        for (n <- 0 to 10) {
+          queue.enqueue(n.doubleValue())
+        }
+
+        var updatedEv: ElectricVehicle =
+          evNoQueue.updateChargingPricesMemory(queue)
+        updatedEv.chargingPricesMemory shouldBe queue
+
+        for (n <- 11 to 25) {
+          queue.enqueue(n.doubleValue())
+        }
+
+        updatedEv = updatedEv.updateChargingPricesMemory(queue)
+        updatedEv.chargingPricesMemory shouldBe queue.drop(7)
+      }
```

Comment:
Since the queues are now immutable, test now does not work anymore.

Please use `(1 to 20).to(immutable.Queue)` or the like to set up the test queues.